Language/Java/Transformer.hs

```diff
@@ -0,0 +1,391 @@
+  where
+    identFromTryResource (TryResourceVarDecl (ResourceDecl _ _ vardDeclId _)) = Just (identFromVarDeclId vardDeclId)
+    identFromTryResource _ = Nothing
+localVarsInStmt _ = []
+
+-- boiler plate cases
+
+instance Transform CompilationUnit where
+  analyze scope (CompilationUnit mbPackageDecl importDecls typeDecls) =
+    CompilationUnit
+      mbPackageDecl
+      importDecls
+      (map (analyze scope) typeDecls)
+
+instance Transform TypeDecl where
+  analyze scope (ClassTypeDecl classDecl) = ClassTypeDecl (analyze scope classDecl)
+  analyze scope (InterfaceTypeDecl interfaceDecl) = InterfaceTypeDecl (analyze scope interfaceDecl)
+
+instance Transform InterfaceDecl where
+  analyze scope (InterfaceDecl src kind modifiers ident typeParams refTypesExtends refTypesPermits interfaceBody) =
+    InterfaceDecl
+      src
+      kind
+      (map (analyze scope) modifiers)
+      ident
+      typeParams
+      refTypesExtends
+      refTypesPermits
+      (analyze scope interfaceBody)
+
+instance Transform Modifier where
+  analyze _ (Public src) = Public src
+  analyze _ Private = Private
+  analyze _ Protected = Protected
+  analyze _ (Abstract src) = Abstract src
+  analyze _ Final = Final
+  analyze _ Static = Static
+  analyze _ StrictFP = StrictFP
+  analyze _ Transient = Transient
+  analyze _ Volatile = Volatile
+  analyze _ Native = Native
+  analyze scope (Annotation anno) = Annotation (analyze scope anno)
+  analyze _ Synchronized_ = Synchronized_
+  analyze _ Sealed = Sealed
+
+instance Transform Annotation where
+  analyze scope (NormalAnnotation span annName annkV) =
+    NormalAnnotation
+      span
+      annName
+      (map (second (analyze scope)) annkV)
+  analyze scope (SingleElementAnnotation span annName annValue) =
+    SingleElementAnnotation span annName (analyze scope annValue)
+  analyze _ (MarkerAnnotation span annName) = MarkerAnnotation span annName
+
+instance Transform ClassBody where
+  analyze scope (ClassBody decls) = ClassBody (map (analyze scope) decls)
+
+instance Transform EnumBody where
+  analyze scope (EnumBody enumConstants decls) = EnumBody (map (analyze scope) enumConstants) (map (analyze scope) decls)
+
+instance Transform InterfaceBody where
+  analyze scope (InterfaceBody memberDecls) = InterfaceBody (map (analyze scope) memberDecls)
+
+instance Transform ElementValue where
+  analyze scope (EVVal varInit) = EVVal (analyze scope varInit)
+  analyze scope (EVAnn anno) = EVAnn (analyze scope anno)
+
+instance Transform Decl where
+  analyze scope (MemberDecl memberDecl) = MemberDecl (analyze scope memberDecl)
+  analyze scope (InitDecl static block) = InitDecl static (analyze scope block)
+
+instance Transform EnumConstant where
+  analyze scope (EnumConstant ident arguments mbClassBody) = EnumConstant ident (map (analyze scope) arguments) (fmap (analyze scope) mbClassBody)
+
+instance Transform VarInit where
+  analyze scope (InitExp exp) = InitExp (analyze scope exp)
```

Comment:
Hatten wir nicht besprochen, dass du _Shadowing_ vermeidest? Hier kannst du einfach `expr` statt `exp` nutzen.

Reply:
Dachte wir beseitigen die Probleme gleichzeitig in allen Datein, wenn ich fertig bin. Damit die Benennung ebenmäßig bleibt.

Reply:
Ok mach ich.